e2e/testcases/multi_sync_test.go

```diff
@@ -574,14 +577,35 @@ func TestConflictingDefinitions_RootToRoot(t *testing.T) {
 	nt.T.Logf("Stop the admission webhook, to confirm that the remediator still reports the conflicts")
 	nomostest.StopWebhook(nt)
 
-	nt.T.Logf("The Role resource version should be changed because two reconcilers are fighting with each other")
-	err = nt.Watcher.WatchObject(kinds.Role(), "pods", testNs,
-		[]testpredicates.Predicate{
-			testpredicates.ResourceVersionNotEquals(nt.Scheme, roleResourceVersion),
-		})
-	if err != nil {
-		nt.T.Fatal(err)
-	}
+	// Watch Role until the other conditions are met.
+	// Collect data from role updates for validation later
+	// This will log diffs for easier debugging.
+	var roleUpdates []roleUpdate
+	nt.T.Log("Starting Role Watcher")
+	bgCtx, bgCancel := context.WithCancel(nt.Context)
+	defer bgCancel()
+	bgDone := make(chan struct{})
+	go func() {
```

Comment:
could this be part of the taskgroup below?

Reply:
The tasks group tasks exit when the predicates are satisfied. This one stays running until cancelled, after the task group is done.

I tried putting this one in the task group, but it requires either two channels to close or an atomic counter to figure out when the other two are done, which felt messy. I don't really know if this is better, but it's at least obvious that it's different.

Reply:
Oh, the other reason it can't be in the TaskGroup is that the Role watcher needs to be _interrupted_ when the TaskGroup is done, because there may not be another Role update to trigger the predicate evaluation.

Reply:
Ack, makes sense